openwisp_controller/config/whois/services.py

```diff
@@ -0,0 +1,103 @@
+from ipaddress import ip_address
+
+from django.core.cache import cache
+from django.db import transaction
+from swapper import load_model
+
+from openwisp_controller.config.settings import WHOIS_ENABLED
+
+from .tasks import fetch_whois_details
+
+
+class WhoIsService:
+    """
+    A handler class for managing the WhoIs functionality.
+    """
+
+    # slots are used to define a fixed set of attributes for the class,
+    # which can improve performance by avoiding the overhead of a dynamic dictionary.
+    # Particularly useful in scenarios where many instances of the class are created.
+    # Reference: https://wiki.python.org/moin/UsingSlots
+    __slots__ = ("device", "org_settings")
+
```

Comment:
If we are creating a cached property, do we need to define slots? I am not sure of their relevance here